src/build/__init__.py

```diff
@@ -49,49 +49,47 @@ class TypoWarning(Warning):
     """
 
 
-class IncompleteCheckWarning(Warning):
+def check_dependency(req_string, ancestral_req_strings=(), parent_extras=frozenset()):
+    # type: (str, Tuple[str, ...], AbstractSet[str]) -> Iterator[Tuple[str, ...]]
     """
-    Warning raised when we have an incomplete check
-    """
-
+    Verify that a dependency and all of its dependencies are met.
 
-def check_version(requirement_string, extra=''):  # type: (str, str) -> bool
+    :param req_string: Requirement string
+    :param parent_extras: Extras (eg. "test" in myproject[test])
+    :yields: Unmet dependencies
     """
-    :param requirement_string: Requirement string
-    :param extra: Extra (eg. test in myproject[test])
-    """
-    import packaging.requirements
+    from packaging.requirements import Requirement
```

Comment:
Please keep the original import, instead of using a partial import.

Reply:
We talked about this earlier and for what it's worth, I'm still strongly -1 on original imports and +1 on partial imports. However 🤷

Reply:
Restored.